llvm/lib/Analysis/IVDescriptors.cpp

```diff
@@ -1081,15 +1086,15 @@ RecurrenceDescriptor::getReductionOpChain(PHINode *Phi, Loop *L) const {
   // more expensive than out-of-loop reductions, and need to be costed more
   // carefully.
   unsigned ExpectedUses = 1;
-  if (RedOp == Instruction::ICmp || RedOp == Instruction::FCmp)
+  if (!IsArithmetic)
```

Comment:
Sounds like IsArithmetic needs to be inverted, since we're checking the negation everywhere, but then I wouldn't know what to call the new variable.

Reply:
We can call it as IsNonArithmetic, I think.
82a107d4b172d22f132b49febe4f819efd686327